### PR TITLE
[Placeholder] BWC Tests for DLS/FLS

### PR DESCRIPTION
**Pull request by parasjain1**
### Description
Updated BWCs and need them to run in GitHub CI actions. This is for the issue mentioned below.

### Issues Resolved
https://github.com/opensearch-project/security/issues/3053

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by scrawfor99** [comment]
Going to convert this to a draft while debugging is in progress. Thank you for your efforts @parasjain1

**Comment by cwperks** [comment]
FYI I ran the bwc tests with previous version set to 3.0.0 and before the cluster reboot, the setup works to ingest data and search before the cluster reboot.

```
    [2023-09-25T17:00:40,138][INFO ][o.o.s.b.SecurityBackwardsCompatibilityIT] [testDataIngestionAndSearchBackwardsCompatibility] HTTP/2.0 200 OK
    [2023-09-25T17:00:40,196][INFO ][o.o.s.b.SecurityBackwardsCompatibilityIT] [testDataIngestionAndSearchBackwardsCompatibility] GET _plugins/_security/api/internalusers/user
     {"user":{"hash":"","reserved":false,"hidden":false,"backend_roles":[],"attributes":{},"opendistro_security_roles":["test-dls-fls-role"],"static":false}}
    [2023-09-25T17:00:40,605][INFO ][o.o.s.b.SecurityBackwardsCompatibilityIT] [testDataIngestionAndSearchBackwardsCompatibility] GET _plugins/_security/api/roles/test-dls-fls-role
     {"test-dls-fls-role":{"reserved":false,"hidden":false,"cluster_permissions":["unlimited"],"index_permissions":[{"index_patterns":["test_index*"],"dls":"{\n        \"bool\": {\n                \"must\": {\n                        \"match\": {\n                                \"genre\": \"rock\"\n                        }\n                }\n        }\n }","fls":["~lyrics"],"masked_fields":["artist"],"allowed_actions":["read","write"]}],"tenant_permissions":[],"static":false}}
    [2023-09-25T17:00:40,605][INFO ][o.o.s.b.SecurityBackwardsCompatibilityIT] [testDataIngestionAndSearchBackwardsCompatibility] after : user exists: true, role exists: true
    [2023-09-25T17:00:41,499][INFO ][o.o.s.b.SecurityBackwardsCompatibilityIT] [testDataIngestionAndSearchBackwardsCompatibility] Ingested data
    [2023-09-25T17:00:42,097][INFO ][o.o.s.b.SecurityBackwardsCompatibilityIT] [testDataIngestionAndSearchBackwardsCompatibility] after test
```

The failures looks like it may be specific to the actual requests being performed after cluster reboot.

File: bwc-test/src/test/java/org/opensearch/security/bwc/SecurityBackwardsCompatibilityIT.java [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+
+public class SecurityBackwardsCompatibilityIT extends OpenSearchRestTestCase {
+
+    private ClusterType CLUSTER_TYPE;
+    private String CLUSTER_NAME;
+
+    private static final String TEST_USER = "user";
+    private static final String TEST_PASSWORD = UUID.randomUUID().toString();
```
Comment (cwperks) [inline bwc-test/src/test/java/org/opensearch/security/bwc/SecurityBackwardsCompatibilityIT.java:67]: Hey @parasjain1, I'm not entirely sure why just yet but its this line that is the problem you are currently facing. I have faced problems with static variables and testClusters in the past. Can you hardcode this to something like `testPassword123_`?

After updating this pw, I got another error:

```
org.opensearch.security.bwc.SecurityBackwardsCompatibilityIT > testDataIngestionAndSearchBackwardsCompatibility FAILED
    org.opensearch.client.ResponseException: method [POST], host [https://127.0.0.1:40011/], URI [test_index-second/_search], status line [HTTP/1.1 404 Not Found]
    {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [test_index-second]","index":"test_index-second","resource.id":"test_index-second","resource.type":"index_or_alias","index_uuid":"_na_"}],"type":"index_not_found_exception","reason":"no such index [test_index-second]","index":"test_index-second","resource.id":"test_index-second","resource.type":"index_or_alias","index_uuid":"_na_"},"status":404}
```

Hopefully this helps you get unstuck.

**Comment by peternied** [comment]
@parasjain1 I'm going to close out this PR since your contribution included these changes - thanks!

**Pull request closed without merging**